### ISSUE TITLE
[Feature]: Add `port` config option from node-ssh to Baremetal config

### ISSUE THREAD

**Issue by cannikin** [redwoodjs/redwood#6426]
### Summary

Forward the `port` config option from our `deploy.toml` config file to the underlying node-ssh lib.

### Motivation

Being able to SSH into a machine on a port other than 22 isn't currently possible.

### Detailed proposal

The list of options are here: https://github.com/redwoodjs/redwood/blob/main/packages/cli/src/commands/deploy/baremetal.js#L635-L644 

We just need to add `port` to that list!

### Are you interested in working on this?

- [X] I'm interested in working on this

**Comment by Josh-Walker-GM** [comment]
The generated default `deploy.toml` includes the following header:

```
# This file contains config for a baremetal deployment
#
# SSH connection options include:
#
# * host - the remote server hostname/IP
# * port - defaults to 22
# * username - required, the user you're connecting as
# * password - only set if you're not using key-based authentication
# * privateKey - a Buffer containing the private key (use this _or_ 'privateKeyPath', not both)
# * privateKeyPath - local file path to a private key that will be sent with the connection request
# * passphrase - used if your private key has a passphrase
# * agentForward - set to `true` to forward the client machine's ssh credentials
#
# See https://redwoodjs.com/docs/deploy/baremetal for more info
```

I hope now the port option will be passed through and used. Do we need to check the others also or are they already passed through?

We may also need to update the documentation at [https://redwoodjs.com/docs/deploy/baremetal#configuration](https://redwoodjs.com/docs/deploy/baremetal#configuration) to mention the port config option?

Again this is my first open source contribution so thanks for your patience.

**Comment by Josh-Walker-GM** [comment]
Solved in #6428 😄
